src/gamecontroller/gamecontrollertrigger.cpp

```diff
@@ -35,12 +38,27 @@ const GameControllerTrigger::ThrottleTypes GameControllerTrigger::DEFAULTTHROTTL
 
 GameControllerTrigger::GameControllerTrigger(int index, int originset, SetJoystick *parentSet, QObject *parent)
     : JoyAxis(index, originset, parentSet, parent)
+    , m_haptic_feedback(0)
 {
     naxisbutton->deleteLater();
     naxisbutton = new GameControllerTriggerButton(this, 0, originset, parentSet, this);
     paxisbutton->deleteLater();
     paxisbutton = new GameControllerTriggerButton(this, 1, originset, parentSet, this);
     reset(index);
+
+#if SDL_VERSION_ATLEAST(2, 0, 16)
+    if (parentSet->getInputDevice()->getControllerType() == SDL_GameControllerType::SDL_CONTROLLER_TYPE_PS5)
+        m_haptic_feedback = new HapticFeedback();
+#endif
+}
+
+GameControllerTrigger::~GameControllerTrigger()
+{
+    if (m_haptic_feedback)
+    {
+        delete m_haptic_feedback;
+        m_haptic_feedback = 0;
+    }
```

Comment:
Followup to previous comment. In case of using `QObject` as a base for `HapticFeedback` this part would be skippable

Reply:
Fixed in next patchset.